auth/user_mgt.go

```diff
@@ -56,6 +56,37 @@ type UserInfo struct {
 	UID        string `json:"rawId,omitempty"`
 }
 
+// MultiFactorInfoResponse describes the `mfaInfo` of the user record API response
+type MultiFactorInfoResponse struct {
+	MFAEnrollmentID string `json:"mfaEnrollmentId"`
+	DisplayName     string `json:"displayName"`
+	PhoneInfo       string `json:"phoneInfo"`
+	EnrolledAt      string `json:"enrolledAt"`
+}
+
+// MultiFactorID represents the type of an enrolled factor, for now only Phone
+// is available.
+type MultiFactorID string
+
+const (
+	// Phone represents an enrolled factor of type Phone / SMS
+	Phone MultiFactorID = "phone"
+)
+
+// PhoneMultiFactorInfo describes a user enrolled second phone factor.
+type PhoneMultiFactorInfo struct {
+	UID                 string
+	DisplayName         string
+	EnrollmentTimestamp int64
+	FactorID            MultiFactorID
+	PhoneNumber         string
+}
+
+// MultiFactorSettings describes the multi-factor related user settings.
+type MultiFactorSettings struct {
+	EnrolledFactors []*PhoneMultiFactorInfo
```

Comment:
This cannot be `PhoneMultiFactorInfo`. We'd want to support other auth types in the future, and at that point we'll have to break this API. Let's just rename `PhoneMultiFactorInfo` struct to `MultiFactorInfo`. Then we can add new fields to that type when we introduce a new auth factor, and that won't be a breaking change.